plugins-ui/src/modules/policy/context/PolicyProvider.tsx

```diff
@@ -60,10 +60,15 @@ export const PolicyProvider: React.FC<{ children: React.ReactNode }> = ({
   };
 
   const removePolicy = async (policyId: string): Promise<void> => {
-    await PolicyService.deletePolicy(policyId);
-    const updatedPolicyMap = new Map(policyMap);
-    updatedPolicyMap.delete(policyId);
-    setPolicyMap(updatedPolicyMap);
+    const policy = policyMap.get(policyId);
+
+    if (policy) {
+      await signPolicy(policy);
```

Comment:
We can return the signature, so the FE knows if the signing gets rejected
(in the case signing is not rejected there will be BE validation of the signature)